### PR TITLE
Add two arguments for ES's Jenkins

### PR DESCRIPTION
**Pull request by nik9000**
`--reference` adds support for using a local mirror which is how
Elastic's CI infrastructure usually speeds up big clones.

`--rely_on_ssh_auth` relies on the SSH authentication already configured
to perform all tasks which is conventient because Elastic's CI already
*has* SSH authentication configured and so do many users. Mechanically
causes the script to skip the git authentication tests entirely,
presuming that SSH authentication is already configured. It also causes
the script to transform all configured repositories that happen to use
`https` into `ssh`.

### CODE REVIEW

File: lib/ES/Repo.pm [hunk 1]
```diff
@@ -60,7 +70,7 @@ sub update_from_remote {
         unless ( $self->_try_to_fetch ) {
```
Comment (nik9000) [inline lib/ES/Repo.pm]: Right now if we don't check that the reference is the same as the last execution in `_try_to_fetch`. We probably should update it if it has changed. That isn't super likely, but it is probably worth doing.

File: build_docs.pl [hunk 2]
```diff
@@ -390,24 +392,28 @@ sub init_repos {
     my $tracker_path = $Conf->{paths}{branch_tracker}
         or die "Missing <paths.branch_tracker> in config";
 
+    my $reference_dir = dir($Opts->{reference})->absolute;
+    die "Missing reference directory $reference_dir" unless -e $reference_dir;
```
Comment (ddillinger) [inline build_docs.pl]: Two things:

1. Maybe I am missing some context (or my perl is just too rusty) but this seems like it's a pretty non-optional option if it's going to `die` on this?
2. Looking at the diff I am guessing it was already like that but what on earth happened to this sub's indentation? It is wrong in the PR diff view, but _looks_ right in raw view, which makes me suspect tabs/spaces mixing is going on maybe?

**Comment by nik9000** [comment]
> By which mainly I mean "does it produce an error that is helpful enough to a human who wants to fix it"?

It'll die with the output of the failed git command. That seems ok to me.

File: build_docs.pl (hunk 2 above)
Comment (nik9000) [reply build_docs.pl]: RE indentation: Bleh! It looks right locally which is super odd.

RE non-optional: I'll fix it! That'd have broken everything..... I need to test this some more too.

**Comment by nik9000** [comment]
Thanks for reviewing @ddillinger!

**Pull request merged**